test/powershell/engine/Basic/Credential.Tests.ps1

```diff
@@ -0,0 +1,6 @@
+Describe "Credential tests" -Tags "CI" {
+    It "Explicit cast for an empty credential returns null" {
```

Comment:
Should we add a case for an empty username?  How do you get the password from the object if `GetNetworkCredential` returns `$null`

Reply:
PSCredential constructor fails if username is empty. The only way to get an empty credentials is [PSCredential]::Empty. And if we can [PSCredential]::Empty any methods shouldn't throw for the empty credential - we're just fix the issue.

Reply:
Actually, you are able to create empty PSCredential instances with a public constructor `PSCredential(psobject pso)`
```powershell
$psobj = [pscustomobject] @{}
$cred = [pscredential]::new($psobj)
```
The current implementation of `GetNetworkCredential()` doesn't allow returning a null value -- it throws if the user name is invalid. I think we should keep it that way and throw exception when user name is null or empty.

Reply:
Why we allow the sample? It seems constructor should throw in the case.
On the other hand, [PSCredential]::Empty is a valid object - why any its method should throw?

Reply:
> Why we allow the sample?

Do you mean the XML comment? I missed the comment, and it looks totally fine to return null then :)

Reply:
I meant your sample:
```
$psobj = [pscustomobject] @{}
$cred = [pscredential]::new($psobj)
```
Why the constructor don't throw?  It seems the constructor should throw.

Reply:
I don't know the original intention, but changing this behavior would be a breaking change for sure -- it makes something that works stop working. It feels insignificant to me, so I don't think it's worth to be fixed. But you can go both ways -- it's insignificant so a breaking change might not matter.

Reply:
For the PR I think we can leave the constructor as is.

Please clarify what is your conclusion about the PR fix?

Reply:
Based on the existing XML comments for `GetNetworkCredential()`, I withdraw my comment about making `GetNetworkCredential` throw. Returning null is totally fine.